src/mount.rs

```diff
@@ -0,0 +1,95 @@
+use nickel::Nickel;
+use request::Request;
+use response::Response;
+use middleware::{Continue, Middleware, MiddlewareResult};
+
+use hyper::uri::RequestUri::AbsolutePath;
+
+pub trait Mountable {
+    fn mount<M: Middleware>(&mut self, mount_point: &str, middleware: M);
+}
+
+impl Mountable for Nickel {
+    fn mount<M: Middleware>(&mut self, mount_point: &str, middleware: M) {
+        self.utilize(Mount::new(mount_point, middleware));
+    }
+}
+
+pub struct Mount<M: Middleware> {
+    mount_point: String,
+    middleware: M
+}
+
+fn rewrite_mount_point(mount_point: &str) -> String {
+    let mut mount_point = mount_point.to_owned();
+    if let Some('/') = mount_point.chars().last() {
+        let _ = mount_point.pop();
+    }
+    if let Some('/') = mount_point.chars().nth(0) {
+        mount_point
+    } else {
+        mount_point.insert(0, '/');
+        mount_point
+    }
+}
+
+impl<M: Middleware> Mount<M> {
+    ///
+    /// Creates a new middleware that mounts a middleware at a mount point.
+    /// An incoming request that matches the mount point will be forwareded to
+    /// the mounted middleware, but with the path rewritten so that the mount
+    /// point appears to be the root from the perspective of the mounted
+    /// middleware. This can be useful in combination with the
+    /// `StaticFilesMiddleware`, for example.
+    ///
+    ///
+    /// # Examples
+    /// ```{rust}
+    /// use nickel::{Nickel, StaticFilesHandler, Mount};
+    /// let mut server = Nickel::new();
+    ///
+    /// server.utilize(
+    ///     Mount::new("/static_files/",
+    ///                StaticFilesHandler::new("/path/to/serve/")
+    /// ));
+    /// ```
+    pub fn new(mount_point: &str, middleware: M) -> Mount<M> {
+        Mount {
+            /* 
+             * Ensure that the mount_point string begins with a / but does not end
+             * with one.
+             */
+            mount_point: rewrite_mount_point(mount_point),
+            middleware: middleware
+        }
+    }
+}
+
+impl<M: Middleware> Middleware for Mount<M> {
+    fn invoke<'mw, 'conn>(&'mw self, req: &mut Request<'mw, 'conn>, res: Response<'mw>)
+        -> MiddlewareResult<'mw> {
+        if let AbsolutePath(path) = req.origin.uri.clone() {
```

Comment:
Of course! I never thought to return early in the non mathing case! This will probably work and looks nicer than what I could think of :)